nuxeo-distribution/nuxeo-wizard-tests/src/test/java/org/nuxeo/ftest/wizard/ITWizardAndUpdateCenterTests.java

```diff
@@ -248,7 +249,13 @@ public void registerConnectInstance() throws InterruptedException {
         String windowHandle = driver.getWindowHandle();
 
         Locator.findElementWithTimeoutAndClick(By.cssSelector(".btn-submit"));
-        Locator.waitUntilWindowClosed(windowHandle);
+        try {
+            Locator.waitUntilWindowClosed(windowHandle);
+        } catch (TimeoutException e) {
+            log.info("Unable to close SSO form once; try again");
+            Locator.findElementWithTimeoutAndClick(By.cssSelector(".btn-submit"));
+            Locator.waitUntilWindowClosed(windowHandle);
+        }
```

Comment:
Is it a known behavior that it fails to close once and succeeds on the second try?
Is it related to the timeout?

Reply:
It only happens on Slaves two times out of three, and, as you can see; it's the window close timeout that expires because of the popup still opened.
A second submit workarounds this strange behavior.